test/algorithms/physical_design/wiring_reduction.cpp

```diff
@@ -0,0 +1,152 @@
+
+#include <fiction/algorithms/physical_design/orthogonal.hpp>
+#include <fiction/algorithms/physical_design/wiring_reduction.hpp>
+#include <fiction/layouts/cartesian_layout.hpp>
+#include <fiction/layouts/clocked_layout.hpp>
+#include <fiction/layouts/coordinates.hpp>
+#include <fiction/layouts/gate_level_layout.hpp>
+#include <fiction/layouts/tile_based_layout.hpp>
+#include <fiction/networks/technology_network.hpp>
+
+#include <mockturtle/networks/aig.hpp>
+#include <mockturtle/views/names_view.hpp>
+
+using namespace fiction;
+
+template <typename Lyt, typename Ntk>
+void check_layout_equiv(const Ntk& ntk)
+{
+    const auto layout = orthogonal<Lyt>(ntk, {});
+
+    wiring_reduction_stats stats{};
+    wiring_reduction<Lyt>(layout, &stats);
+
+    check_eq(ntk, layout);
+
+    CHECK(mockturtle::to_seconds(stats.time_total) > 0);
+}
+
+template <typename Lyt>
+void check_layout_equiv_all()
+{
+    SECTION("Maj1 Network")
+    {
+        check_layout_equiv<Lyt>(blueprints::maj1_network<mockturtle::aig_network>());
+    }
+    SECTION("unbalanced_and_inv_network")
+    {
+        check_layout_equiv<Lyt>(blueprints::unbalanced_and_inv_network<mockturtle::aig_network>());
+    }
+    SECTION("and_or_network")
+    {
+        check_layout_equiv<Lyt>(blueprints::and_or_network<technology_network>());
+    }
+    SECTION("nary_operation_network")
+    {
+        check_layout_equiv<Lyt>(blueprints::nary_operation_network<technology_network>());
+    }
+    SECTION("constant_gate_input_maj_network")
+    {
+        check_layout_equiv<Lyt>(blueprints::constant_gate_input_maj_network<technology_network>());
+    }
+    SECTION("half_adder_network")
+    {
+        check_layout_equiv<Lyt>(blueprints::half_adder_network<technology_network>());
+    }
+    SECTION("full_adder_network")
+    {
+        check_layout_equiv<Lyt>(blueprints::full_adder_network<technology_network>());
+    }
+    SECTION("mux21_network")
+    {
+        check_layout_equiv<Lyt>(blueprints::mux21_network<technology_network>());
+    }
+    SECTION("se_coloring_corner_case_network")
+    {
+        check_layout_equiv<Lyt>(blueprints::se_coloring_corner_case_network<technology_network>());
+    }
+    SECTION("clpl")
+    {
+        check_layout_equiv<Lyt>(blueprints::clpl<technology_network>());
+    }
+    SECTION("nand_xnor_network")
+    {
+        check_layout_equiv<Lyt>(blueprints::nand_xnor_network<technology_network>());
+    }
+    SECTION("fanout_substitution_corner_case_network")
+    {
+        check_layout_equiv<Lyt>(blueprints::fanout_substitution_corner_case_network<technology_network>());
+    }
+    SECTION("inverter_network")
+    {
+        check_layout_equiv<Lyt>(blueprints::inverter_network<technology_network>());
+    }
+}
+
+TEST_CASE("Layout equivalence", "[wiring_reduction]")
+{
+    SECTION("Cartesian layouts")
+    {
+        using gate_layout = gate_level_layout<clocked_layout<tile_based_layout<cartesian_layout<offset::ucoord_t>>>>;
```

Comment:
Would you mind adding tests with `cube::coord_t` coordinates as well?

Reply:
Doing so leads to a lot of errors where I don't think they are caused `wiring_reduction`.
For example:

fiction/layouts/gate_level_layout.hpp:255:16:
error: cannot convert 'const tile' (aka 'const fiction::cube::coord_t') to 'signal' (aka 'unsigned long long') without a conversion operator
**return static_cast<signal>(t);**

Which comes from the creation of the initial layout:

/utils/blueprints/network_blueprints.hpp:187:25:
const auto x1 = ntk.create_pi();